### PR TITLE
Set associations between classes

### PR DESCRIPTION
**Pull request by Al-Amin-Khan-Shakil**
### In this pull request I have implemented the has-many and belongs to relations between student and classroom class and rental, book, and person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -11,6 +12,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (skyv26) [inline person.rb:15]: - [ ] @Al-Amin-Khan-Shakil you did undoubtedly amazing job, but you missed a very small thing while writing the code, you have added the `instance` variable for adding `rentals` but you didn't provide the window for the user to `use` the `filter` to retrieve the data or **add** more rental data. So to resolve the problem, you just need to add the `getter` and `setter` for the **rentals**.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require './rental'
+
+# Book class
+class Book
+  attr_accessor :title, :author
```
Comment (skyv26) [inline book.rb]: - [ ] @Al-Amin-Khan-Shakil please do the same, what I mentioned in my above comment related to `rentals`.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @Al-Amin-Khan-Shakil, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Implemented the features ✅ 
- Professional `README` template ✅ 

## Required Changes ♻️

- [OPTIONAL] Please try to add more descriptive PR, you added a good description but you can make it more better and informative which tells what specifically you achieved, did or learnt while developing the milestone. 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by LionRouge1 — APPROVED** [review]
Hi @Al-Amin-Khan-Shakil, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Highlights
- Great job, your project is error free :+1: 
- Nice work you meet all the requirements :1st_place_medal: 
- All linters passed :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**